conjure-java-jackson-optimization/src/main/java/com/palantir/conjure/java/optimizations/ObjectMapperOptimizations.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.optimizations;
+
+import com.fasterxml.jackson.module.blackbird.BlackbirdModule;
+import java.util.List;
+
+/**
+ * Applies jackson optimization modules based on supported JVMs and best practices.
+ * This allows us to quickly, transparently change between implementations such as Afterburner, Blackbird, or none at
+ * all.
+ */
+public final class ObjectMapperOptimizations {
+
+    private static final boolean NO_OPTIMIZATIONS =
+            // These optimizations are not used within graalvm because they leverage dynamic class creation
+            System.getProperty("org.graalvm.nativeimage.imagecode") != null
+                    // This may be globally disabled with a system property
+                    || "true".equalsIgnoreCase(System.getProperty("com.palantir.conjure.java.optimizations.disabled"));
```

Comment:
Is optimizations too generic? What if you have some other "optimizations" in conjure that you'd like to disable?

Reply:
Basically all I'm suggesting is "com.palantir.conjure.java.optimizations.disabled" -> "com.palantir.conjure.java.jackson.optimizations.disabled", but mostly a nit

Reply:
Oh sorry, I misinterpreted your comment! Updating the package name and disable string totally make sense 👍

Reply:
Not sure how I got it in my head that the comment referred to the class name `ObjectMapperOptimizations`. Updated!